Occasional failure in TestInscribedRectangle::test_inscribed_rectangle

## Describe the bug

TestInscribedRectangle::test_inscribed_rectangle occasionally fails and crashes the CI. This is due to the random nature of the test, and the fact that there are some rather rare edge cases which are not properly accounted for.

## Steps to reproduce

Steps to reproduce the behaviour:

Run the tests locally or on the CI and get unlucky.

## Expected behaviour

The test should always pass, and the edge cases shouldn't fail the test.

## Screenshots / Tracebacks

```python
=========================== short test summary info ============================
FAILED tests/bluemira/geometry/test_inscribed_rect.py::TestInscribedRectangle::test_inscribed_rectangle[shape3-False]
= 1 failed, 306 passed, 18 skipped, 8 deselected, 2 warnings in 134.10s (0:02:14) =
Error: Process completed with exit code 1.
```

## Environment (please complete the following information)

CI 

## Additional context

Add any other context about the problem here.

--- a/tests/BLUEPRINT/geometry/test_inscribedrect.py
+++ b/tests/BLUEPRINT/geometry/test_inscribedrect.py
@@ -65,12 +65,13 @@ class TestInscribedRectangle:
         for i in range(x):
             for j in range(y):
                 point = points[:, i, j]
-                if shape.point_in_poly(point):
+                if shape.point_in_poly(point, include_edges=False):
                     for k in self.aspectratios:
                         dx, dz = inscribed_rect_in_poly(
                             shape, point, aspectratio=k, convex=convex
                         )
                         sq = _rect(point[0], point[1], dx, dz)
+                        assert len(sq.x) == 5
                         try:
                             tf = boolean_2d_difference(sq, shape)
                         except ValueError:
@@ -81,12 +82,14 @@ class TestInscribedRectangle:
                             sq.plot(ax, linewidth=0.1)
 
                         if tf is not None:
-                            self.assertion_error_creator(
-                                "Overlap", [dx, dz, point, k, convex]
-                            )
-                            if tests.PLOTTING:
-                                for t in tf:
-                                    t.plot(ax, facecolor="r", linewidth=0.1)
+                            if not all([t.area == 0.0 for t in tf]):
+
+                                self.assertion_error_creator(
+                                    "Overlap", [dx, dz, point, k, convex]
+                                )
+                                if tests.PLOTTING:
+                                    for t in tf:
+                                        t.plot(ax, facecolor="r", linewidth=0.1)
                         if not np.allclose(dx / dz, k):
                             self.assertion_error_creator("Aspect", [dx, dz, dx / dz, k])
 

